contracts/MysteryBox.sol

```diff
@@ -35,15 +35,18 @@ contract MysteryBox is OwnableUpgradeable {
         uint32 personal_limit;
         uint32 start_time;
         uint32 end_time;
+        bool canceled;
         // sell all NFT(s) owned
         bool sell_all;
         uint256[] nft_id_list;
         address qualification;
         mapping(address => uint256[]) purchased_nft;
         // total number of NFT(s)
         uint256 total;
-        bool canceled;
-        // `mask holder`
+
+        // `token address`
+        address holder_token_addr;
+        // `token holder`
         uint256 holder_min_token_amount;
     }
```

Comment:
I wonder if we could re-order the variable to save gas?

Reply:
I tested it.  No differnce.
`address holder_token_addr` will take one storage slot.

Reply:
you are right. Code updated.